### PR TITLE
#158626942 api implement endpoint 

### PR DESCRIPTION
**Pull request by SelaDanti**
What does this PR do?
This PR request add functionality to the endpoints

Description of Task to be completed?
Implemented the following endpoints
GET /rides
GET /rides/
POST /rides
POST /rides//requests
modified code to based on pep8 guidelines
Added some more test

How the app should be tested
Clone the repo to your local machine
install a virtual enviroment
install the dependacies
from terminal run manage.py
test the endpoint using postman

Any background context you want to provide?
Developed using python and Flask

What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2179557

### CODE REVIEW

File: app/models/rides.py [hunk 1]
```diff
@@ -1,62 +1,222 @@
...
+		elif input['email'].isspace():
+			return True
+		elif input['date'].isspace():
+			return True
+		else:
+			return False
+
+	def correct_data(self,input):
+		months = ['january', 'jebruary', 'march', 'april', 'may', 'june', 'july', 'august',
+		 'september', 'october', 'november', 'december']
```
Comment (JoshuaOndieki) [inline app/models/rides.py]: - On dates, consider [Python Datetime](https://docs.python.org/3/library/datetime.html) module.
- I don't think it's a good idea to hardcode this.
- For example, you have an `elif` referring to 2018. Don't you think this project should be valid in 2019?

**Comment by JoshuaOndieki** [comment]
I suggest you add `--cover-package` flag to your .travis.yml . It will increase your coverage by ignoring dependency packages that you don't need to test.

**Comment by artorious** [comment]
Kudos on progress. Looks good

File: app/models/rides.py [hunk 2]
```diff
@@ -1,62 +1,222 @@
 class rides:
-	def __init__(self):
```
Comment (justmesam) [inline app/models/rides.py]: Why did you delete the `__init__`???

File: app/models/rides.py [hunk 3]
```diff
@@ -1,62 +1,222 @@
...
-		self.all_request = []
-	def add_ride(self,username,car_capacity,contact,time,date,month,year,destination,pickup):
-		data = {
-		'username':username,
-		'car capacity':car_capacity,
-		'contact':contact,
-		'time':time,
-		'date':date,
-		'month':month,
+	'class to relating to rides'
```
Comment (justmesam) [inline app/models/rides.py]: How does this class achieve `OOP`?

File: app/models/rides.py [hunk 4]
```diff
@@ -1,62 +1,222 @@
...
+	ride_data = []
+	request_data = []
+	return_data = None
+
+class new_ride(rides):
```
Comment (justmesam) [inline app/models/rides.py]: All class names should be in capitalized and use camelcase naming convection..

File: app/models/rides.py [hunk 5]
```diff
@@ -1,62 +1,222 @@
...
+print(obj.add_ride('kwame','capacity','time','date','month','year','pickup','destination','email'))
+print(obj.view_ride())
+print(obj.ride_exist(1))
+print(obj.view_ride_single(1))
+print(obj.key_format({'username':'username'}))
+print(obj.same_direction('as','as'))
+print(obj.is_empty(' '))
+print(obj.only_space(' '))
+#print(obj.is_string({'time':'234'}))
+print(obj.is_integer({'capacity':1,'year':2}))"""
```
Comment (justmesam) [inline app/models/rides.py]: Its bad practise to push commented code

File: app/views/views.py [hunk 6]
```diff
@@ -1,62 +1,93 @@
...
-    'pick up point': fields.String("pick up point"),
-    'Destination': fields.String("destination"),
-    'contact': fields.String("contact")
-    })
-ride_request = api.model('ride_request',{
-    'name': fields.String,
-    'pickup time': fields.String,
-    'destination': fields.String,
-    'pick up point': fields.String,
+obj = new_ride()
```
Comment (justmesam) [inline app/views/views.py]: This is not how OOP is used to create instances of `new_ride()`  we use it to create many instances of whatever we are trying to achieve but not a single instance used all over. Or you can take me through your logic of implementation...

File: test/test_rides.py [hunk 7]
```diff
@@ -0,0 +1,615 @@
...
+		response = self.test.post('/rides',
+			data=json.dumps(self.post_ride),headers=self.headers)
+		self.assertEqual(response.status_code,201)
+		self.assertIn('Content Uploaded',response.data)
+
+	def test_post_keys(self):
+		"""
+		test for when posting with the wrong keys
+		"""
+		post_ride = {
```
Comment (justmesam) [inline test/test_rides.py]: create one `post_ride ` on setUp and re-use it rather than repeating yourself..

File: app/models/rides.py (hunk 2 above)
Comment (SelaDanti) [reply app/models/rides.py]: I decided to use the attributes declared in the initialise method in the add method instead. So there was no need of initialising them in the contractor

File: test/test_rides.py (hunk 7 above)
Comment (SelaDanti) [reply test/test_rides.py]: I wanted to test different occurrence of post_ride eg post_ride where post_ride['month'] = 'not a month' or where post_ride['month'] = '      '
